plone/app/layout/viewlets/common.py

```diff
@@ -324,23 +333,21 @@ def navtree(self):
             query["exclude_from_nav"] = False
 
         context_path = "/".join(self.context.getPhysicalPath())
+        self.customize_query(query)
         portal_catalog = getToolByName(self.context, "portal_catalog")
         brains = portal_catalog.searchResults(**query)
 
-        registry = getUtility(IRegistry)
-        types_using_view = registry.get("plone.types_use_view_action_in_listings", [])
-
         for brain in brains:
             brain_path = brain.getPath()
             brain_parent_path = brain_path.rpartition("/")[0]
-            if brain_parent_path == navtree_path:
+            if self.portal_tabs and brain_parent_path == navtree_path:
                 # This should be already provided by the portal_tabs_view
                 continue
             if brain.exclude_from_nav and not context_path.startswith(brain_path):
                 # skip excluded items if they're not in our context path
                 continue
             url = brain.getURL()
-            if brain.portal_type in types_using_view:
+            if brain.portal_type in self.types_using_view:
```

Comment:
For performance I would store self.types_using_view (and maybe above self.portal_tabs) in a local variable before the loops.

Reply:
I thought the expression is resolved once in a loop.
Maybe this is different for ``@properties``?
Changed.

Reply:
done